src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -206,6 +207,18 @@ private async Task<Location> RelocateSymbol_NoLockAsync(Workspace workspace, Met
             return await MetadataAsSourceHelpers.GetLocationInGeneratedSourceAsync(symbolId, temporaryDocument, cancellationToken).ConfigureAwait(false);
         }
 
+        public bool ShouldCollapseOnOpen(string filePath, BlockStructureOptions blockStructureOptions)
+        {
+            if (_generatedFilenameToInformation.TryGetValue(filePath, out var info))
```

Comment:
is it possible for this to return false?

Reply:
Yes, there are multiple providers, and the file being asked about could have been generated by a different one.